pkg/registry/apis/provisioning/jobs/pullrequest.go

```diff
@@ -99,40 +107,51 @@ func NewPullRequestCommenter(
 	}, nil
 }
 
-func (c *PullRequestCommenter) Process(ctx context.Context, job provisioning.Job) error {
-	if job.Spec.Action != provisioning.JobActionPullRequest {
-		return errors.New("job is not a pull request")
-	}
+func (c *pullRequestCommenter) ProcessPullRequest(ctx context.Context,
```

Comment:
I am really not liking this change in the signature as it entangles even more the processing with the status report but we can reorganize things in the next milestone.